packages/integration-tests/test-utils.ts

```diff
@@ -0,0 +1,14 @@
+export function injectOriginToBlankPostMessages(window: Window, origin: string): void {
+  // workaround for https://github.com/jsdom/jsdom/issues/2745
+  // if no origin exists, replace with the wallet url
+  window.addEventListener('message', (event: MessageEvent) => {
+    if (event.origin === '') {
+      event.stopImmediatePropagation();
+      const eventWithOrigin: MessageEvent = new MessageEvent('message', {
+        data: event.data,
+        origin
+      });
+      window.dispatchEvent(eventWithOrigin);
+    }
+  });
+}
```

Comment:
For now this will need to be called in each test file's `beforeAll` block. A nicer way to hide this shim from test authors would be to integrate it into a test environment for jest. https://jestjs.io/docs/en/configuration.html#testenvironment-string